usage/src/org/apache/cloudstack/quota/QuotaManagerImpl.java

```diff
@@ -0,0 +1,465 @@
+                            break;
+                        case QuotaTypes.VOLUME:
+                            quotalistforaccount.add(updateQuotaDiskUsage(usageRecord, aggregationRatio, QuotaTypes.VOLUME));
+                            break;
+                        case QuotaTypes.VM_SNAPSHOT:
+                            quotalistforaccount.add(updateQuotaDiskUsage(usageRecord, aggregationRatio, QuotaTypes.VM_SNAPSHOT));
+                            break;
+                        case QuotaTypes.LOAD_BALANCER_POLICY:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.LOAD_BALANCER_POLICY));
+                            break;
+                        case QuotaTypes.PORT_FORWARDING_RULE:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.PORT_FORWARDING_RULE));
+                            break;
+                        case QuotaTypes.IP_ADDRESS:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.IP_ADDRESS));
+                            break;
+                        case QuotaTypes.NETWORK_OFFERING:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.NETWORK_OFFERING));
+                            break;
+                        case QuotaTypes.SECURITY_GROUP:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.SECURITY_GROUP));
+                            break;
+                        case QuotaTypes.VPN_USERS:
+                            quotalistforaccount.add(updateQuotaRaw(usageRecord, aggregationRatio, QuotaTypes.VPN_USERS));
+                            break;
+                        case QuotaTypes.NETWORK_BYTES_RECEIVED:
+                            quotalistforaccount.add(updateQuotaNetwork(usageRecord, QuotaTypes.NETWORK_BYTES_RECEIVED));
+                            break;
+                        case QuotaTypes.NETWORK_BYTES_SENT:
+                            quotalistforaccount.add(updateQuotaNetwork(usageRecord, QuotaTypes.NETWORK_BYTES_SENT));
+                            break;
+                        case QuotaTypes.VM_DISK_IO_READ:
+                        case QuotaTypes.VM_DISK_IO_WRITE:
+                        case QuotaTypes.VM_DISK_BYTES_READ:
+                        case QuotaTypes.VM_DISK_BYTES_WRITE:
+                        default:
+                            break;
+                        }
+                    }
+                } while ((usageRecords != null) && !usageRecords.first().isEmpty());
+                // list of quotas for this account
+                s_logger.info("Quota entries size = " + quotalistforaccount.size() + ", accId" + account.getAccountId() + ", domId" + account.getDomainId());
+                if (quotalistforaccount.size() > 0) { // balance to be processed
```

Comment:
Consider using ``List#isEmpty()`` instead of checking ``size`` as it is more idiomatic and expressive of intent.